modules/FIC_Mercurial.py

```diff
@@ -99,73 +84,43 @@ def _load_changesets_data_in_json(self):
-        for commit in range(0, len(self.changeset.get("changesets"))):
-            self.commits = self.changeset.get("changesets")[commit]
-            return self.commits
-
-    # Gets author from commits
-    def _get_author(self):
-        self.commit_author = self.commits.get("author")
-        return self.commit_author
-
-    # Gets description from commits
-    def _get_description(self):
-        self.commit_message = self.commits.get("desc")
-        return self.commit_message
-
-    # Gets files changed from commit
-    def _get_changed_files(self):
-        self.commit_files_changed = [file for file in self.commits.get("files")]
-        return self.commit_files_changed
-
-    # Get hash, laster used to format commit url
-    def _get_node(self):
-        self.commit_sha = self.commits.get("node")
-        return self.commit_sha
-
     # Generated commit url by triming hash
-    def _generate_commit_url(self):
-        self.commit_url = self.repository_url + "pushloghtml?changeset=" + self.commit_sha[:12]
+    def _generate_commit_url(self, node):
+        self.commit_url = self.repository_url + "pushloghtml?changeset=" + node[:12]
         return self.commit_url
 
-    # Changeset number
-    def _get_changeset_number(self):
-        for key in self.changesets_json.get("pushes"):
-            self.changeset_number = key
-            return self.changeset_number
-
-    # Get changeset commits as list
-    def _get_changesets_commits(self):
-        for key in self.changesets_json.get("pushes").get(self.changeset_number):
-            self.changeset_commits = self.changesets_json.get("pushes").get(self.changeset_number).get(key)
-            return self.changeset_commits
-
-    # Construct commit dictionary per changeset
-    def _construct_commits_dict(self):
+# Construct json dictionary
+    def _construct_json_dict(self):
         self.hg_commits_list = {}
-        for commit in range(len(self.changeset_commits)):
-            self.hg_commits_list.update({commit + 1: {"url": self.commit_url,
-                                                      "commiter_author": self.commit_author,
-                                                      "commit_message": self.commit_message,
-                                                      "files_changed": self.commit_files_changed}})
-        return self.hg_commits_list
+        self.final_dict = {}
+        for push in self.changesets_json.get("pushes"):
+            unformated_date = self.changesets_json.get("pushes").get(push).get("date")
+            pusher = self.changesets_json.get("pushes").get(push).get("user")
+            date = self._generate_changeset_date(unformated_date)
+            self.final_dict.update({push: {"pusher": pusher,
+                                           "date_of_push": date,
+                                           "changeset_commits": {}}})
+
+            for commit in range(len(self.changesets_json.get("pushes").get(push).get("changesets"))):
```

Comment:
Create method `_get_push_changesets()` from `len(self.changesets_json.get("pushes").get(push).get("changesets"))`.
Also since we are using just the len() to get the range for the `for`, we should specify that it returns the number of changesets so the method should be called `_get_push_changesets_number()`